### DISCUSSION TITLE
Possibility of using both MQTT and HTTPS protocols for the same device identity

### DISCUSSION THREAD

**Discussion by ghost** [Azure/azure-iot-sdk-c#2465] [Q&A] [answered]
I am currently exploring ways of implementing _Upload File to Blob_ features to an already existing application using this SDK on an IoT device. My application uses MQTT protocol and has _Plug and Play_ support which as I understand is exclusive to only MQTT and MQTT-WS protocols while _Upload File to Blob_ feature is supported only by HTTPS.   

I am looking at the possibility of uploading files to Azure as well using [Device Update for IoT Hub](https://learn.microsoft.com/en-us/azure/iot-hub-device-update/understand-device-update) which I'm guessing would need to use some form of file download to my device.  

Now my idea to use features that are available both to MQTT and HTTPS protocols would be to use _two_ `IOTHUB_DEVICE_CLIENT_HANDLE`s for the same device identity, as an example, if I were to use SASTokens the underlying code would look something like this:
```
const char* connectionString = "[device connection string]";

IOTHUB_DEVICE_CLIENT_HANDLE device_handle_mqtt;
IOTHUB_DEVICE_CLIENT_HANDLE device_handle_https;

device_handle_mqtt = IoTHubDeviceClient_CreateFromConnectionString(connectionString, MQTT_Protocol);
device_handle_https = IoTHubDeviceClient_CreateFromConnectionString(connectionString, HTTP_Protocol);
```
In the above example I would use `device_handle_mqtt` for sending telemetry data, _Plug and Play_ features and when I would get a request to download/upload files, I would instead use `device_handle_https`. I hope this explains clearly enough what I am trying to accomplish.  

Would this approach make any sense or is there be a better alternative or perhaps none at all?

**Comment by ericwolz** [comment, marked as answer]
Sorry for the late response. You can not have two simultaneous connections to the hub service at the same time with the same device ID, so that will not work.

There should be no need to add the HTTP_Protocol as file upload should work with the MQTT_Protocol. Although our sample shows using HTTP_Protocol that should not be the requirement as MQTT hub connection should also work.

**Comment by ericwolz** [comment]
I updated the sample to demonstrate usage of MQTT and upload to blob.

#2482
